src/starks/proof/options.rs

```diff
@@ -0,0 +1,211 @@
+use lambdaworks_math::field::traits::IsPrimeField;
+
+use super::errors::InsecureOptionError;
+
+pub enum SecurityLevel {
+    Conjecturable80Bits,
+    Conjecturable100Bits,
+    Conjecturable128Bits,
+    Provable80Bits,
+    Provable100Bits,
+    Provable128Bits,
+}
+
+/// The options for the proof
+///
+/// - `blowup_factor`: the blowup factor for the trace
+/// - `fri_number_of_queries`: the number of queries for the FRI layer
+/// - `coset_offset`: the offset for the coset
+/// - `grinding_factor`: the number of leading zeros that we want for the Hash(hash || nonce)
+#[derive(Clone, Debug)]
+pub struct ProofOptions {
+    pub blowup_factor: u8,
+    pub fri_number_of_queries: usize,
+    pub coset_offset: u64,
+    pub grinding_factor: u8,
+}
+
+impl ProofOptions {
+    const NUM_SECURITY_BITS: usize = 128;
+    // TODO: Make it work for extended fields
+    const EXTENSION_DEGREE: usize = 1;
+    // Estimated maximum domain size. 2^40 = 1 TB
+    const NUM_BITS_MAX_DOMAIN_SIZE: usize = 40;
+
+    /// See section 5.10.1 of https://eprint.iacr.org/2021/582.pdf
+    pub fn new_secure(security_level: SecurityLevel, coset_offset: u64) -> Self {
+        match security_level {
+            SecurityLevel::Conjecturable80Bits => ProofOptions {
+                blowup_factor: 4,
+                fri_number_of_queries: 31,
+                coset_offset,
+                grinding_factor: 20,
+            },
+            SecurityLevel::Conjecturable100Bits => ProofOptions {
+                blowup_factor: 4,
+                fri_number_of_queries: 41,
+                coset_offset,
+                grinding_factor: 20,
+            },
+            SecurityLevel::Conjecturable128Bits => ProofOptions {
+                blowup_factor: 4,
+                fri_number_of_queries: 55,
+                coset_offset,
+                grinding_factor: 20,
+            },
+            SecurityLevel::Provable80Bits => ProofOptions {
+                blowup_factor: 4,
+                fri_number_of_queries: 80,
+                coset_offset,
+                grinding_factor: 20,
+            },
+            SecurityLevel::Provable100Bits => ProofOptions {
+                blowup_factor: 4,
+                fri_number_of_queries: 104,
+                coset_offset,
+                grinding_factor: 20,
+            },
+            SecurityLevel::Provable128Bits => ProofOptions {
+                blowup_factor: 4,
+                fri_number_of_queries: 140,
+                coset_offset,
+                grinding_factor: 20,
+            },
+        }
+    }
+
+    /// Checks security of proof options given 128 bits of security
+    pub fn new_with_checked_security<F: IsPrimeField>(
+        blowup_factor: u8,
+        fri_number_of_queries: usize,
+        coset_offset: u64,
+        grinding_factor: u8,
+        _security_target: u8,
```

Comment:
Is it ok that this input is ignored? Seems like it should be important for the function